server/ServerApp.cpp

```diff
@@ -1869,6 +1867,20 @@ int ServerApp::AddPlayerIntoGame(const PlayerConnectionPtr& player_connection, i
                 break;
             }
         }
+        // Assign player to empire if he doesn't have own empire and delegates signle
```

Comment:
meaning unclear... assign the new player to an empire if that player doesn't already have an empire "and delegates signle" ...?  Maybe "Assign the newly-connected player to an empire that doesn't already have a player and that has delegated (only) the name of the new player to play it"....?

Why can't an empire be delegated to multiple players, and the server assign any of those players to that empire when the player joins?

What if multiple empires have delegated the same player to play them? How does it pick which empire to assign the player to? That seems like the more problematic situation to address...

Reply:
> meaning unclear... assign the new player to an empire if that player doesn't already have an empire "and delegates signle" ...? Maybe "Assign the newly-connected player to an empire that doesn't already have a player and that has delegated (only) the name of the new player to play it"....?

No, this function either chooses single empire for player so he enter to the game and controls that empire or doesn't choose any so player is sent to lobby. Before this PR player who doesn't own empire but have one empire he delegates was sent to lobby where he had choose this empire. AI could not manage it. So after this PR such player automatically assigned to that empire.

> Why can't an empire be delegated to multiple players, and the server assign any of those players to that empire when the player joins?

Yes, it possible but server doesn't search player for the empire, it searches empire for the player.

> What if multiple empires have delegated the same player to play them? How does it pick which empire to assign the player to? That seems like the more problematic situation to address...

In this case player isn't assigned to empire so he is sent to lobby. Then in lobby he choose empire for himself like he did it was open saved game. This function is called again with chosen empire id in `target_empire_id` and else branch checks and assigns chosen empire for the player.